staging/src/k8s.io/component-base/logs/api/v1/options.go

```diff
@@ -117,7 +134,30 @@ func (c *LoggingConfiguration) Validate(featureGate featuregate.FeatureGate, fld
 		}
 	}
 
-	// Currently nothing to validate for c.Options.
+	errs = append(errs, c.validateFormatOptions(featureGate, fldPath.Child("options"))...)
+	return errs
+}
+
+func (c *LoggingConfiguration) validateFormatOptions(featureGate featuregate.FeatureGate, fldPath *field.Path) field.ErrorList {
+	errs := field.ErrorList{}
+	errs = append(errs, c.validateJSONOptions(featureGate, fldPath.Child("json"))...)
+	return errs
+}
+
+func (c *LoggingConfiguration) validateJSONOptions(featureGate featuregate.FeatureGate, fldPath *field.Path) field.ErrorList {
+	errs := field.ErrorList{}
+	enabled := false
+	if featureGate != nil {
+		enabled = featureGate.Enabled(LoggingAlphaOptions)
+	}
+	if !enabled {
```

Comment:
this structure might be correct now, but it's likely that new options will get inserted and will accidentally pick up beta status of other options

I'd suggest structuring this in a way that makes it clear for each option whether it is alpha or beta, something like:

```go
// default to alpha and beta features being disabled
enabled := map[featuregate.Feature]bool{}
if featureGate != nil {
  enabled[LoggingAlphaOptions] = featureGate.Enabled(LoggingAlphaOptions)
  enabled[LoggingBetaOptions] = featureGate.Enabled(LoggingBetaOptions)
}

// make sure alpha and beta config options aren't specified if features aren't enabled

if f := LoggingAlphaOptions; c.Options.JSON.SplitStream && !enabled[f] {
  errs = append(errs, field.Forbidden(fldPath.Child("splitStream"), fmt.Sprintf("Feature %s is disabled", f)))
}
if f := LoggingAlphaOptions; c.Options.JSON.InfoBufferSize.Value() && !enabled[f] {
  errs = append(errs, field.Forbidden(fldPath.Child("infoBufferSize"), fmt.Sprintf("Feature %s is disabled", f)))
}
```

Reply:
I've used a helper function in each if check.